Sources/Routing/Parameter/Parameter.swift

```diff
@@ -7,12 +7,13 @@ import Foundation
 ///         return "user id: \(id)"
 ///     }
 ///
-/// Use the static `parameter` property to generate a `PathComponent` for this type.
+/// Uspublic e the static `parameter` property to generate a `PathComponent` for this type.
 public protocol Parameter {
     /// The type this parameter will convert to once it is looked up.
     /// Most types like `String` and `Int` will simply return self, but some
     /// more complex types may wish to perform async lookups or conversions to different types.
     associatedtype ResolvedParameter
+    associatedtype ParameterType: LosslessDataConvertible
```

Comment:
This associated type indicates the expected type of parameter. It's needed to differentiate from the conforming type. For example

```swift
// matches /foo/10
// The conforming type (Likely `ResolvedParameter`) will be the same as `ParameterType`
"foo", Int.parameter

// matches /foo/1 where User.ID == Int
// The conforming type is different than the `ParameterType`
"foo", User.parameter